src/main/java/com/opmining/MainProgram.java

```diff
@@ -85,21 +102,43 @@ public static void main(String[] args) throws UnsupportedEncodingException, IOEx
                 try {
                     JSONParser parser = new JSONParser();
                     JSONObject data = (JSONObject)parser.parse(cont);
+                    String testDate = (String)data.get("testDate");
+                    String testerId = (String)data.get("testerId");
+                    String deviceName = (String)data.get("deviceName");
+                    String companyName = (String)data.get("companyName");
                     String cont1 = (String)data.get("content");
                     statExtractor.setText(cont1);
                     statExtractor.extractInfo();
                     JSONObject json = new JSONObject();
-                    json.put("Text",cont1);
-                    json.put("Sentence Counts",statExtractor.getSentenceCnt());
-                    json.put("Word Counts",statExtractor.getWordCnt());
-                    json.put("Positive Words",statExtractor.getPositiveWordCnt());
-                    json.put("Negative Words",statExtractor.getNegativeWordCnt());
-                    publisher2.publish("statCommentStream",json.toString());
+                    json.put("companyName",companyName);
+                    json.put("deviceName",deviceName);
+                    json.put("testerId",testerId);
+                    json.put("testDate",testDate);
+                    json.put("content",cont1);
+                    json.put("sentenceCount",statExtractor.getSentenceCnt());
+                    json.put("wordCount",statExtractor.getWordCnt());
+                    json.put("positiveWordCount",statExtractor.getPositiveWordCnt());
+                    json.put("negativeWordCount",statExtractor.getNegativeWordCnt());
+                    publisher2.publish("commentStatStream",json.toString());
+                     // Let's write the statistical information to the database         
+                    op.update(new BasicDBObject("content",cont1).append("testerId",testerId),
```

Comment:
update kafamı karıştırdı. insert olması gerekmez miydi? Kafka'dan veriyi alıyoruz, json üzerinde update yapıp (vtye yazmadan) en son update edilmiş json objesini (istatistiksel bilgi kısımları) Mongo'ya insert etmemiz gerekmez miydi ? (önce insert sonra update yerine)

Reply:
update kısmının Cuma günü, sunucu da düzeltebiliriz, şimdilik kendi bilgisayarlarımızda doğru çalışabilmesi ve test edilebilmesi için Update gerekiyor.